runtime/amd64.S

```diff
@@ -1121,6 +1121,7 @@ LBL(do_perform):
         TSAN_RESTORE_CALLER_REGS
         LEAVE_FUNCTION
 #endif
+        movq    %rdi, 8(%rbx) /* Set the last fiber field in the continuation */
         movq    Stack_handler(%rsi), %r11  /* %r11 := old stack -> handler */
```

Comment:
Are these runtime changes to last fiber field a bug fix? They seem unrelated to the effect syntax changes.

Reply:
These are related to the syntax change and not a bug fix. These changes were needed due to https://github.com/ocaml/ocaml/pull/12735, which stores both ends of the continuation in the continuation object. 

In particular, in the function based effect API, the effect handler function sets the last fiber.

https://github.com/ocaml/ocaml/pull/12735/files#diff-4ed190e806a9a1c8ba763ca078b1643d77a11194beda581ac35ee80b282089b9R83

It it cumbersome (and not the appropriate place) to replicate this in the compiler-generated code for handling effects in the syntax based version. Hence, I've pushed updating the last fiber logic to the relevant backends and the interpreter (`interp.c`).